### ISSUE TITLE
Let tokenizers be installed via Pip

### ISSUE THREAD

**Issue by bittner** [nltk/nltk#3413]
Call me stupid, but why do NLTK users [have to download NLTK corpora](https://www.nltk.org/data.html) using Python code?

This is not only non-standard, but causes a lot of technical and usability problems along the road (see [related issues](https://github.com/nltk/nltk/issues?q=is%3Aissue%20download) and [StackOverflow](https://stackoverflow.com/questions/77131746/how-to-download-punkt-tokenizer-in-nltk)). Software shouldn't download package data in a custom, self-crafted way when there is PyPI that offers a (more) secure and battle-tested infrastructure.

Isn't it possible to include [the tokenizers](https://www.nltk.org/nltk_data/) via [more `extras_require` targets](https://github.com/nltk/nltk/blob/develop/setup.py#L32C1-L32C15) or separate Python packages instead?

**Comment by ryanamannion** [comment]
I agree it would be helpful. Looking at discussions in the past for this, it looks like the cited reason has been for 
  
  1. lack of resources [[link to comment](https://github.com/nltk/nltk/issues/55#issuecomment-236334927)]

  2. Maximum file size on pypi 
     * The max is [currently 100 MiB](https://github.com/pypi/warehouse/blob/58dc95e9f38e895e5c7cb203a99076bf98a64b65/warehouse/constants.py#L5). As far as I can tell, the largest nltk_data package is Framenet v1.7 at 94.6 MiB. Dataset sizes here: https://www.nltk.org/nltk_data/

The most relevant issues can be found here:
* https://github.com/nltk/nltk/issues/55
* https://github.com/nltk/nltk_data/issues/12

I think spaCy gets around the pypi file size limit by having their models installable via their downloader `$ python -m spacy download en_core_web_sm`, which installs the wheel that they host on github e.g.

```
$ python -m spacy download en_core_web_sm
Looking in indexes: https://pypi.org/simple
Collecting en-core-web-sm==3.7.1
  Downloading https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-3.7.1/en_core_web_sm-3.7.1-py3-none-any.whl (12.8 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.8/12.8 MB 18.3 MB/s eta 0:00:00
...
Installing collected packages: en-core-web-sm
Successfully installed en-core-web-sm-3.7.1
✔ Download and installation successful
You can now load the package via spacy.load('en_core_web_sm')
$ pip freeze | grep en_core_web_sm
en-core-web-sm @ https://github.com/explosion/spacy-models/releases/download/en_core_web_sm-3.7.1/en_core_web_sm-3.7.1-py3-none-any.whl#sha256=86cc141f63942d4b2c5fcee06630fd6f904788d2f0ab005cce45aadb8fb73889
```

More info here: https://github.com/explosion/spacy-models

**Comment by ekaf** [comment]
While distributing a minimal “essentials” NLTK data package via pip would be highly desirable and offer a great out-of-the-box experience, unfortunately the unclear or restrictive licensing terms of several key datasets (such as punkt, stopwords, names, brown, etc.) make such a package impossible to distribute through PyPI at present. 

There have been suggestions that some resources, like Punkt, could be “rescued” and re-released under a standard permissive license, but the legal basis for this seems weak and remains unresolved. The most commonly needed NLTK data resources lack FOSS-compliant or even clearly permissive licenses, so bundling them for automated installation would violate PyPI’s requirements and present legal risks.

For now, users must continue to download these datasets manually or via NLTK’s downloader, which leaves responsibility for license compliance with each user.

Still, a minimal "essentials" NLTK data package could provide a great out-of-the-box experience. Here’s a suggested selection that covers most common tutorials and use-cases, without being too large. Unfortunately, many of these have unclear licenses:

| Package Name                 | Purpose / Example Use                 |
|------------------------------|---------------------------------------|
| punkt                        | Tokenization (sentences/words)        |
| averaged_perceptron_tagger   | Part-of-speech tagging                |
| stopwords                    | Filtering stopwords                   |
| wordnet                      | Lemmatization, word relations         |
| omw-1.4                      | WordNet lemmatizer, multilingual data |
| snowball_data                | Stemming (various languages)          |
| names                        | Classification demos                  |
| brown or movie_reviews       | Corpus for demos, exploration         |

**Rationale:**
- Tokenization and tagging are required for almost every NLTK tutorial.
- Stopwords and lemmatization are essential for text preprocessing.
- Corpora like Brown or movie_reviews allow users to quickly experiment with real data.
- `names` is tiny but enables classic NLTK classifier demos.
- `wordnet` (and omw-1.4) are needed for lemmatization, synonyms, and semantic tasks.

This selection is small (tens of MBs, not hundreds), covers 90% of beginner and intermediate NLTK use-cases, and enables the official tutorials to work out-of-the-box.

**Comment by ekaf** [comment]
### **Proposal: A Two-Package Solution for pip Distribution**

The goal of providing an easy pip-installable data package is complicated by significant licensing uncertainties concerning _model provenance_ because the pre-trained models (`punkt`, `averaged_perceptron_tagger`, `maxent_treebank_pos_tagger`, `maxent_ne_chunker`) lack explicit licenses. Their status is ambiguous because their training data provenance is undocumented and likely based on corpora with non-commercial license restrictions (e.g., LDC data).

Also, while _semcor_  can circumvent the restrictions on the Brown Corpus, we may doubt that the Princeton WordNet team actually had the rights to relicense the Brown Corpus under a more permissive license, and would need to confirm it.

**Why PyPI Allows Restricted Packages:**
PyPI's primary role is as a package repository, not a license enforcer. It requires a license classifier but does not mandate that packages be open source. Its policies focus on security and metadata, placing the responsibility for license compliance on the package maintainer (to label correctly) and the end-user (to obey the terms). This is why classifiers like `"License :: Free for non-commercial use"` exist.

### **Proposed Pip Packages**

To resolve this, let's build two separate packages:

#### **1. `nltk-essentials` (Commercial-Safe, < 100 MB)**
**License:** Various Permissive (MIT, Apache 2.0, CC-BY, Public Domain)
**Content:** Data with unambiguous, commercial-friendly licenses.
*   **Lexicons/Lists:** `vader_lexicon`, `stopwords`
*   **Semantic Resources:** `wordnet`, `omw-1.4`
*   **Utilities:** `universal_tagset`, `tagsets` (*Excludes `perluniprops` due to license ambiguity*)
*   **Corpora (Public Domain):** `genesis`, `inaugural`, a `gutenberg` subset

#### **2. `nltk-data-research` (Restricted, < 100 MB)**
**Classifier:** `"License :: Free for non-commercial use"`
**Content:** Pre-trained models and data with restrictive or unclear licenses.
*   **Models:** `punkt_tab`, `averaged_perceptron_tagger`, `maxent_treebank_pos_tagger`, `maxent_ne_chunker`
*   **Corpora:** `semcor`
*   **Utilities:** `perluniprops` (*Included here due to its ambiguous licensing*)

This structure provides a safe, pip-installable core (`nltk-essentials`) for all users, while making advanced models available to academia via a clearly labeled, restricted package. Combining these two packages should provide a satisfactory experience for most users.

**Comment by stevenbird** [comment]
I for one welcome an overhaul here. It would be good to have broad consensus about a good way forward.

**Comment by ekaf** [comment]
A smooth path forward could start by defining two collections in nltk_data, listing the contents of the two suggested PyPi bundles. The "essentials" bundle would correspond almost to the existing "popular" collection.
